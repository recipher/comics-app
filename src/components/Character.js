import React, { memo } from 'react';
import { Link } from '@reach/router';

const Character = memo(({ id, name, description, image }) => {
  return (
    <div>
      <Link to={`/comics/${id}`}><h3>{name}</h3></Link>
      <p dangerouslySetInnerHTML={{ __html: description }}/>
      <img src={image} alt={description} />
      <hr/>
    </div>
  );
});

export default Character;
