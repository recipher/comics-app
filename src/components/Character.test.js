import React from 'react';
import 'react-testing-library/cleanup-after-each';
import { render } from 'react-testing-library';
import Character from './Character';

const CHARACTER = {
  id: 1234,
  name: 'Captain Marvel',
  description: '...',
  image: 'image.jpg',
};

it('should match snapshot', () => {
  const { container } = render(<Character {...CHARACTER} />);

  expect(container).toMatchSnapshot();
});
