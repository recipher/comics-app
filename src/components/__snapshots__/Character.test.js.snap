// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div>
    <a
      href="/comics/1234"
    >
      <h3>
        Captain Marvel
      </h3>
    </a>
    <p>
      ...
    </p>
    <img
      alt="..."
      src="image.jpg"
    />
    <hr />
  </div>
</div>
`;
